napari/layers/_tests/test_layer_actions.py

```diff
@@ -37,13 +38,25 @@ def _dummy():
 )
 def test_projections(mode):
     ll = LayerList()
-    ll.append(Image(np.random.rand(8, 8, 8)))
+    ll.append(Image(np.random.rand(6, 7, 8)))
     assert len(ll) == 1
     assert ll[-1].data.ndim == 3
-    _project(ll, mode=mode)
+    dims = Dims(ndim=ll[-1].data.ndim, order=(0, 1, 2))
+    _project(ll, dims=dims, mode=mode)
     assert len(ll) == 2
     # because keepdims = False
-    assert ll[-1].data.shape == (8, 8)
+    assert ll[-1].data.shape == (7, 8)
+
+    # Test swapping axes for proper projection visualization with image visual.
+    ll.selection.active = ll[0]
```

Comment:
Would be great to split these out into multiple tests to cover the different orders, so you can avoid any unwanted/unexpected side effects (e.g. changing the active selection). That way you also avoid the need to reset the selection as on this line.

This test is already parametrized, so I think separate tests is probably the way to go here - that also makes it easier to see the relationship between order and the resulting shape.